airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/activities/GenerateInputActivityImpl.java

```diff
@@ -104,105 +108,220 @@ private void saveAttemptSyncConfig(final long jobId, final int attemptNumber, fi
         "set attempt sync config");
   }
 
+  private IntegrationLauncherConfig getSourceIntegrationLauncherConfig(final long jobId,
```

Comment:
This whole part about building config feels duplicated (from `DefaultJobCreator` maybe?). Is it possible to share this code with the current version to avoid duplication?

Reply:
Hmm, IntegrationLauncherConfig for syncs are only built here - I don't see this in DefaultJobCreator. Could you clarify which duplication you're referring to?